src/node_file.cc

```diff
@@ -2106,6 +2106,51 @@ static void OpenFileHandle(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+// TODO(@anonrig): Implement v8 fast APi calls for `cpSync`.
+static void CpSync(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args.Length() ==
+        8);  // src, dest, preserveTimestamps, dereference, errorOnExist, force,
+             // recursive, verbatimSymlinks
+  BufferValue src(env->isolate(), args[0]);
+  CHECK_NOT_NULL(*src);
+  ToNamespacedPath(env, &src);
+
+  BufferValue dest(env->isolate(), args[1]);
+  CHECK_NOT_NULL(*dest);
+  ToNamespacedPath(env, &dest);
+
+  bool preserveTimestamps = args[2]->IsTrue();
+  bool dereference = args[3]->IsTrue();
+  bool errorOnExist = args[4]->IsTrue();
+  bool force = args[5]->IsTrue();
+  bool recursive = args[6]->IsTrue();
+  bool verbatimSymlinks = args[7]->IsTrue();
+
+  if (errorOnExist) {
+    if (std::filesystem::exists(src.ToStringView())) {
+      return env->ThrowUVException(UV_EEXIST, "EEXIST", "File already exists");
+    }
+  }
+
+  std::filesystem::copy_options options =
+      std::filesystem::copy_options::update_existing;
+
+  // When true timestamps from src will be preserved.
+  if (preserveTimestamps)
+    options |= std::filesystem::copy_options::create_hard_links;
+  // Dereference symbolic links.
+  if (dereference) options |= std::filesystem::copy_options::copy_symlinks;
+  // Overwrite existing file or directory.
+  if (force) options |= std::filesystem::copy_options::overwrite_existing;
+  // Copy directories recursively.
+  if (recursive) options |= std::filesystem::copy_options::recursive;
+  // When true, path resolution for symlinks will be skipped.
+  if (verbatimSymlinks) options |= std::filesystem::copy_options::skip_symlinks;
+
+  std::filesystem::copy(src.ToStringView(), dest.ToStringView(), options);
```

Comment:
This can also throw, right? Would be good to make sure these are handling the exceptions.